query/src/main/java/org/infinispan/query/dsl/embedded/impl/BaseIckleFilterIndexingServiceProvider.java

```diff
@@ -298,16 +306,20 @@ public Object getTarget() {
       }
 
       @Override
-      public void invoke(Event<K, V> event) {
+      public CompletionStage<Void> invoke(Event<K, V> event) {
+         return null;
       }
 
       @Override
-      public void invoke(EventWrapper<K, V, CacheEntryEvent<K, V>> event, boolean isLocalNodePrimaryOwner) {
+      public CompletionStage<Void> invoke(EventWrapper<K, V, CacheEntryEvent<K, V>> event, boolean isLocalNodePrimaryOwner) {
          matchEvent(event, matcher);
+         // TODO: make this non blocking at some point?
```

Comment:
Make returning null non blocking? :D

Reply:
is referring to the whole method including matchEvent which does the actual notification

Reply:
Much better :)